👷‍♀️🐶 add workflows and reviewdog

merca opened the pull request:
Problem
No verification workflow exists

Solution
- ⬆ add dependabot for `pip`
- 👮‍♀️ add codeQL
- 💢 add sonarlint
- 👕 add markdown
- 🐶 add reviewdog
- ⚫ add black
- 🔰 add pylint
- 🎂 add mypy

To simulate testing and such
- add api init and method for login with not implemeted error
- ✔ add tests that verify not implemented error

Commit:
👷‍♀️🐶 add workflows and reviewdog (#6)

Problem
No verification workflow exists

Solution
- ⬆ add dependabot for `pip`
- 👮‍♀️ add codeQL
- 💢 add sonarlint
- 👕 add markdown
- 🐶 add reviewdog
- ⚫ add black
- 🔰 add pylint
- 🎂 add mypy

To simulate testing and such
- add api init and method for login with not implemeted error
- ✔ add tests that verify not implemented error

theovald commented on .github/workflows/python_lint.yml:
```diff
@@ -0,0 +1,38 @@
+on:
+  push:
+    branches:
+      - current
+      - next
```
What is the branching naming strategy here, out of curiosity?

merca replied:
@theovald : Trying out new things. `current` for current release (main in conventional word), `next` for next release (dev in conventional word)

theovald commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,34 @@
+      - current
+      - next
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Isnt this just the default? Added for clearity?

merca replied:
Yes, so that I also learn me some defaults 😆

theovald commented on aduro/__init__.py:
```diff
@@ -0,0 +1,26 @@
+
+
+class Aduro:
+    """Aduro Stove proxi client"""
```
```suggestion
    """Aduro Stove proxy client"""
```

theovald commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+dynamic = ["version", "description"]
+
+[project.urls]
+Source = "https://github.com/merca/pyAnduro"
```
```suggestion
Source = "https://github.com/merca/pyAduro"
```

Tenrec commented on aduro/__init__.py:
```diff
@@ -0,0 +1,26 @@
+
+
+class Aduro:
+    """Aduro Stove proxi client"""
```
```suggestion
    """Aduro Stove proxi client."""
```

Tenrec replied:
Doesn't the linting object to this?

merca replied:
@Tenrec strangely not 😢 ... need better linting 🤷‍♀️

Commit:
Apply suggestions from code review

Co-authored-by: Nora <72867025+theovald@users.noreply.github.com>

Commit:
✨ Base stuff (models, tests and such) (#8)

* ✨ Add data models and tests

* ♻ recfactor and resolve 🐶 problems

* 🐶 fix rest of the review dog problems

* 🔰 make pylint happy

* reduce nesting in `try_convert_object`
  - remove nesting, use if/else
  - add some more tests to make sure also `None`will be covered

* switch to `pydantic`

* Typed return for `helpers.try_convert_object`
